Index: src/app/apis/shop-api/index.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/app/apis/shop-api/index.ts	(date 1528511404000)
+++ src/app/apis/shop-api/index.ts	(date 1528512015000)
@@ -19,6 +19,13 @@
       }, 100);
     });
   }
+
+  async updateProduct(product: Product): Promise<void> {
+    await this.put(
+      `products/${product.id}`,
+      product,
+    );
+  }
 }
 
 const shopAPI: ShopAPI = new ShopAPIImpl();
Index: src/app/apis/types.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/app/apis/types.ts	(date 1528511404000)
+++ src/app/apis/types.ts	(date 1528512015000)
@@ -12,6 +12,8 @@
   getProducts(): Promise<Product[]>;
 
   buyProducts(products: Array<{ id: number, quantity: number }>): Promise<void>;
+
+  updateProduct(product: Product): Promise<void>;
 }
 
 //----------------------------------------------------------------------
Index: src/app/stores/product-store/index.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/app/stores/product-store/index.ts	(date 1528511404000)
+++ src/app/stores/product-store/index.ts	(date 1528512015000)
@@ -57,6 +57,14 @@
     this.state.all = products;
   }
 
+  async updateProduct(product: Product): Promise<void> {
+    const stateProduct = this.getStateProductById(product.id);
+    if (stateProduct) {
+      await this.$apis.shop.updateProduct(product);
+      this.$utils.assignIn(stateProduct, product);
+    }
+  }
+
   //----------------------------------------------------------------------
   //
   //  Internal methods
Index: src/app/stores/types.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/app/stores/types.ts	(date 1528511404000)
+++ src/app/stores/types.ts	(date 1528512015000)
@@ -20,6 +20,8 @@
   decrementProductInventory(productId: number): void;
 
   getAllProducts(): Promise<void>;
+
+  updateProduct(product: Product): Promise<void>;
 }
 
 export interface CartStore {
Index: src/app/views/product-admin-view/index.vue
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/app/views/product-admin-view/index.vue	(date 1528511404000)
+++ src/app/views/product-admin-view/index.vue	(date 1528512015000)
@@ -61,6 +61,10 @@
       }
     }
   }
+
+  .update-button {
+    color: var(--app-link-color);
+  }
 </style>
 
 
@@ -136,6 +140,15 @@
             ></paper-input>
           </div>
         </div>
+      </div>
+      <div class="layout horizontal center end-justified">
+        <paper-button
+          v-show="!!selectedProduct.id"
+          class="update-button"
+          @click="updateProduct"
+        >
+          Update
+        </paper-button>
       </div>
     </div>
 
@@ -185,6 +198,16 @@
     created() {
     }
 
+    //----------------------------------------------------------------------
+    //
+    //  Internal methods
+    //
+    //----------------------------------------------------------------------
+
+    async updateProduct(): Promise<void> {
+      await this.$stores.product.updateProduct(this.selectedProduct as Product);
+    }
+
     //----------------------------------------------------------------------
     //
     //  Event handlers
