Index: src/app/views/abc-view/index.vue
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/app/views/abc-view/index.vue	(revision 32dcc44df4a0aa530cbfbbc88981f652e4f660ff)
+++ src/app/views/abc-view/index.vue	(date 1528892179000)
@@ -1,37 +1,44 @@
 <style lang="stylus" scoped>
+  @import '../../../assets/styles/_variables.styl';
+
+  .container {
+    @extend .app-shadow-elevation-2dp;
+    @extend .app-pa-3;
+  }
+
   .greet-message {
-    --greet-message-color: var(--paper-indigo-a200);
+    --greet-message-color: $app-indigo-a700;
   }
 </style>
 
 
 <template>
   <div class="layout vertical">
-    <paper-card class="app-pa-3" :class="{ 'app-ma-12': pc, 'app-ma-6': tab, 'app-ma-3': sp }">
-      <paper-input
-        label="Input Message"
-        :value="message" @input="message = $event.target.value"
-      ></paper-input>
+    <div
+      class="container"
+      :class="{ 'app-ma-12': pc, 'app-ma-6': tab, 'app-ma-3': sp }"
+    >
+      <p>message: <input v-model="message"></p>
       <p>propA: {{ propA }}</p>
       <p>propB: {{ propB }}</p>
       <p>message: {{ message }}</p>
       <p>custom propA: {{ customPropA }}</p>
       <p>reversed message: {{ reversedMessage }}</p>
       <p>double reversed message: {{ doubleReversedMessage }}</p>
-      <div class="layout horizontal center">
+      <p class="layout horizontal center">
         <greet-message
           ref="greetMessage"
           :message="message"
           class="greet-message"
         ></greet-message>
         <button class="app-ml-3" @click="greetButtonOnClick">Greet</button>
-      </div>
+      </p>
       <p class="layout horizontal">
         <custom-input
           v-model="customInputValue"
           class="flex-3"
         ></custom-input>
-        <span class="flex-9 app-ml-3 app-mt-7">value: {{ customInputValue }}</span>
+        <span class="flex-9 app-ml-3">value: {{ customInputValue }}</span>
       </p>
       <p class="layout horizontal center">
         <custom-checkbox
@@ -40,10 +47,10 @@
         <span class="app-ml-3">checked: {{ customChecked }}</span>
       </p>
       <div class="layout horizontal end-justified">
-        <paper-button raised @click="postButtonOnClick">Post</paper-button>
-        <paper-button raised @click="sleepButtonOnClick">Sleep</paper-button>
+        <button @click="postButtonOnClick">Post</button>
+        <button class="app-ml-2" @click="sleepButtonOnClick">Sleep</button>
       </div>
-    </paper-card>
+    </div>
   </div>
 </template>
 
Index: src/assets/styles/_spacing.styl
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/assets/styles/_spacing.styl	(revision 32dcc44df4a0aa530cbfbbc88981f652e4f660ff)
+++ src/assets/styles/_spacing.styl	(date 1528891326000)
@@ -63,10 +63,16 @@
   margin-right: auto !important;
 }
 
-:root {
-  $i = 0
-  for $app-spacer-size in $app-spacer-size-list {
-    --app-spacer-{$i}: $app-spacer-size;
-    $i = $i + 1
-  }
-}
+$app-spacer-0 = $app-spacer-size-list[0];
+$app-spacer-1 = $app-spacer-size-list[1];
+$app-spacer-2 = $app-spacer-size-list[2];
+$app-spacer-3 = $app-spacer-size-list[3];
+$app-spacer-4 = $app-spacer-size-list[4];
+$app-spacer-5 = $app-spacer-size-list[5];
+$app-spacer-6 = $app-spacer-size-list[6];
+$app-spacer-7 = $app-spacer-size-list[7];
+$app-spacer-8 = $app-spacer-size-list[8];
+$app-spacer-9 = $app-spacer-size-list[9];
+$app-spacer-10 = $app-spacer-size-list[10];
+$app-spacer-11 = $app-spacer-size-list[11];
+$app-spacer-12 = $app-spacer-size-list[12];
Index: src/assets/styles/_variables.styl
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/assets/styles/_variables.styl	(date 1528891326000)
+++ src/assets/styles/_variables.styl	(date 1528891326000)
@@ -0,0 +1,29 @@
+@import './_colors.styl';
+@import './_flex-layout.styl';
+@import './_shadows.styl';
+@import './_spacing.styl';
+@import './_typography.styl';
+
+/**
+ * Application theme
+ */
+
+$app-primary-text-color = $app-grey-900;
+$app-secondary-text-color = $app-grey-600;
+$app-disabled-text-color = $app-grey-400;
+$app-accent-text-color = $app-pink-a200;
+$app-error-text-color = $app-red-a700;
+$app-link-color = $app-blue-800;
+$app-link-visited-color = $app-indigo-300;
+$app-default-border-color = $app-grey-300;
+
+
+/**
+ * Grid
+ */
+
+$app-grid-breakpoints-xs = 0;
+$app-grid-breakpoints-sm = 600px;
+$app-grid-breakpoints-md = 960px;
+$app-grid-breakpoints-lg = calc(1280px - 16px); /* Desktop gets a 16dp reduction */
+$app-grid-breakpoints-xl = calc(1920px - 16px); /* https://material.io/guidelines/layout/responsive-ui.html#responsive-ui-breakpoints */
Index: src/app/views/index.vue
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/app/views/index.vue	(revision 32dcc44df4a0aa530cbfbbc88981f652e4f660ff)
+++ src/app/views/index.vue	(date 1528892179000)
@@ -1,5 +1,5 @@
 <style lang="stylus" scoped>
-  @import '../../assets/styles/_typography.styl';
+  @import '../../assets/styles/_variables.styl';
 
   app-drawer-layout {
     --app-drawer-width: 256px;
@@ -8,75 +8,61 @@
     }
   }
 
-  .drawer-toolbar {
+  .container {
+    height 100%;
   }
 
-  .content-toolbar {
-    background-color: var(--paper-indigo-a200);
-    color: #fff;
-  }
+  .drawer {
+    width 256px;
+    background-color: $app-grey-100;
+    border-right: 1px solid $app-default-border-color;
+    .main-title {
+      @extend .app-font-title;
+      @extend .app-pa-5;
+      img {
+        width: 24px;
+        height: 24px;
+      }
+    }
 
-  .drawer-list {
-    .item {
-      display: block;
-      padding: var(--app-spacer-2) var(--app-spacer-5);
-      @extend .app-font-code1;
-      color: var(--app-secondary-text-color);
-      text-decoration none;
-    }
-
-    .item.router-link-active {
-      color: var(--app-accent-text-color);
-    }
-  }
-
-  .link-button {
-    color: var(--app-link-color);
+    .list {
+      .item {
+        display: block;
+        padding: $app-spacer-2  $app-spacer-5;
+        @extend .app-font-code1;
+        color: $app-secondary-text-color;
+        text-decoration none;
+        &.router-link-active {
+          color: $app-accent-text-color;
+        }
+      }
+    }
   }
 </style>
 
 
 <template>
-  <div>
-
-    <app-drawer-layout responsive-width="960px">
-
-      <!--
-        Drawer content
-      -->
-      <app-drawer ref="drawer" slot="drawer" :swipe-open="narrow">
-        <app-toolbar class="drawer-toolbar">
-          <iron-icon src="assets/images/manifest/icon-48x48.png"></iron-icon>
-          <div main-title class="app-ml-2">Vue WWW Base</div>
-        </app-toolbar>
-        <div class="drawer-list">
-          <template v-for="item in items">
-            <router-link :to="item.path" class="item">{{ item.title }}</router-link>
-          </template>
-        </div>
-      </app-drawer>
-
-      <!--
-        Main content
-      -->
-      <app-toolbar class="content-toolbar">
-        <paper-icon-button icon="menu" drawer-toggle></paper-icon-button>
-        <div main-title>View name</div>
-      </app-toolbar>
-
+  <div class="layout horizontal container">
+    <!--
+      Drawer content
+    -->
+    <div class="layout vertical drawer">
+      <div class="layout horizontal center main-title">
+        <img src="assets/images/manifest/icon-48x48.png"/>
+        <div class="app-ml-2">Vue WWW Base</div>
+      </div>
+      <div class="flex list">
+        <template v-for="item in items">
+          <router-link :to="item.path" class="item">{{ item.title }}</router-link>
+        </template>
+      </div>
+    </div>
+    <!--
+      Main content
+    -->
+    <div class="flex">
       <router-view/>
-
-    </app-drawer-layout>
-
-    <paper-toast ref="swToast" :duration="swUpdateIsRequired ? 0 : 5000" :text="swMessage">
-      <paper-button
-        v-show="swUpdateIsRequired"
-        class="link-button"
-        @click="reload"
-      >再読み込み
-      </paper-button>
-    </paper-toast>
-
+    </div>
   </div>
 </template>
 
@@ -87,19 +73,6 @@
 import { ElementComponent } from '../components';
 import { mixins } from 'vue-class-component';
 
-import '@polymer/app-layout/app-drawer-layout/app-drawer-layout';
-import '@polymer/app-layout/app-drawer/app-drawer';
-import '@polymer/app-layout/app-header-layout/app-header-layout';
-import '@polymer/app-layout/app-header/app-header';
-import '@polymer/app-layout/app-toolbar/app-toolbar';
-import '@polymer/iron-icon/iron-icon';
-import '@polymer/iron-icons/iron-icons';
-import '@polymer/iron-pages/iron-pages';
-import '@polymer/iron-selector/iron-selector';
-import '@polymer/paper-button/paper-button';
-import '@polymer/paper-icon-button/paper-icon-button';
-import '@polymer/paper-toast/paper-toast';
-
 @Component
 export default class AppView extends mixins(ElementComponent) {
   //----------------------------------------------------------------------
@@ -108,10 +81,6 @@
   //
   //----------------------------------------------------------------------
 
-  private page: string = '';
-
-  private narrow: boolean = false;
-
   private items: Array<{ title: string; path: string }> = [
     {
       title: 'ABC',
@@ -123,18 +92,6 @@
     },
   ];
 
-  private swMessage: string = '';
-
-  private swUpdateIsRequired: boolean = false;
-
-  //--------------------------------------------------
-  //  Elements
-  //--------------------------------------------------
-
-  private get swToast(): { open: () => void } {
-    return this.$refs.swToast as any;
-  }
-
   //----------------------------------------------------------------------
   //
   //  Lifecycle hooks
@@ -148,16 +105,6 @@
     this.$stores.product.getAllProducts();
   }
 
-  //----------------------------------------------------------------------
-  //
-  //  Internal methods
-  //
-  //----------------------------------------------------------------------
-
-  private reload(): void {
-    window.location.reload();
-  }
-
   //----------------------------------------------------------------------
   //
   //  Event handlers
@@ -165,10 +112,6 @@
   //----------------------------------------------------------------------
 
   private swOnStateChange(info: sw.StateChangeInfo) {
-    this.swMessage = info.message;
-    this.swUpdateIsRequired = info.state === sw.ChangeState.updateIsRequired;
-    this.$nextTick(() => this.swToast.open());
-
     // tslint:disable-next-line
     console.log(info);
   }
Index: src/assets/styles/main.styl
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/assets/styles/main.styl	(revision 32dcc44df4a0aa530cbfbbc88981f652e4f660ff)
+++ src/assets/styles/main.styl	(date 1528891357000)
@@ -1,62 +1,37 @@
 @import url('https://fonts.googleapis.com/css?family=Roboto:300,400,500,700|Material+Icons');
-@import './_colors.styl'
-@import './_flex-layout.styl'
-@import './_shadows.styl'
-@import './_spacing.styl'
-@import './_typography.styl'
+@import './_colors.styl';
+@import './_flex-layout.styl';
+@import './_shadows.styl';
+@import './_spacing.styl';
+@import './_typography.styl';
+@import './_variables.styl';
 
 /*-------------------------------------------------
   Variables
 -------------------------------------------------*/
 
 :root {
-  /**
-   * Application theme
-   */
-
-  --app-primary-text-color: var(--app-grey-900);
-  --app-secondary-text-color: var(--app-grey-600);
-  --app-disabled-text-color: var(--app-grey-400);
-  --app-accent-text-color: var(--app-pink-a200);
-  --app-error-text-color: var(--app-red-a700);
-  --app-link-color: var(--app-light-blue-a400);
-  --app-link-visited-color: var(--app-indigo-300);
-  --app-default-border-color: var(--app-grey-300);
-
-  /**
-   * Grid
-   */
-
-  --app-grid-breakpoints-xs: 0;
-  --app-grid-breakpoints-sm: 600px;
-  --app-grid-breakpoints-md: 960px;
-  --app-grid-breakpoints-lg: calc(1280px - 16px); /* Desktop gets a 16dp reduction */
-  --app-grid-breakpoints-xl: calc(1920px - 16px); /* https://material.io/guidelines/layout/responsive-ui.html#responsive-ui-breakpoints */
 
   /**
    * Components
    */
 
-  /* paper-drawer-panel */
-  --drawer-menu-color: #ffffff;
-  --drawer-border-color: 1px solid var(--appdefault-border-color);
-  --drawer-toolbar-border-color: 1px solid var(--app-default-border-color);
-
-  /* paper-menu */
-  --paper-menu-background-color: #ffffff;
-  --menu-link-color: var(--app-primary-text-color);
-
   /* greet-message */
-  --greet-message-color: var(--app-red-500);
+  --greet-message-color: $app-red-500;
+
 }
 
 /*-------------------------------------------------
   Default style
 -------------------------------------------------*/
 
+html, body {
+  height 100%;
+}
+
 body {
   margin: 0;
   font-family: 'Roboto', 'Noto', sans-serif;
   background-color: #ffffff;
-  color: var(--app-primary-text-color);
+  color: $app-primary-text-color;
 }
Index: src/index.html
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/index.html	(revision 32dcc44df4a0aa530cbfbbc88981f652e4f660ff)
+++ src/index.html	(date 1528891357000)
@@ -38,20 +38,11 @@
   <meta name="msapplication-TileImage" content="assets/images/manifest/icon-144x144.png">
   <meta name="msapplication-TileColor" content="#3f51b5">
   <meta name="msapplication-tap-highlight" content="no">
-
-  <script src="node_modules/@webcomponents/webcomponentsjs/webcomponents-bundle.js"></script>
 </head>
 
 <body>
   <div id="app"></div>
-  <script>
-    window.addEventListener('WebComponentsReady', () => {
-      const script = document.createElement('script');
-      script.onload = () => { /* do stuff with the script */ };
-      script.src = '<%= htmlWebpackPlugin.options.bundledScript %>';
-      document.head.appendChild(script);
-    });
-  </script>
+  <script src="<%= htmlWebpackPlugin.options.bundledScript %>"></script>
 </body>
 
 </html>
\ No newline at end of file
Index: build/webpack.base.conf.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- build/webpack.base.conf.js	(revision 32dcc44df4a0aa530cbfbbc88981f652e4f660ff)
+++ build/webpack.base.conf.js	(date 1528888751000)
@@ -33,7 +33,8 @@
     mode: settings.mode,
 
     entry: {
-      'index': path.resolve(__dirname, '../src/app/index.ts'),
+      // `babel-polyfill`はIE11対応
+      'index': ['babel-polyfill', path.resolve(__dirname, '../src/app/index.ts')],
     },
 
     output: {
Index: src/app/views/abc-view/custom-input.vue
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/app/views/abc-view/custom-input.vue	(revision 32dcc44df4a0aa530cbfbbc88981f652e4f660ff)
+++ src/app/views/abc-view/custom-input.vue	(date 1528892179000)
@@ -1,18 +1,16 @@
 <style lang="stylus" scoped>
+  .text-field {
+    width 100%;
+  }
 </style>
 
-
 <template>
   <div>
-    <paper-input
-      @input="textFieldOnChange"
-    ></paper-input>
+    <input class="text-field" @input="textFieldOnChange"/>
   </div>
 </template>
 
-
 <script lang="ts">
-import '@polymer/paper-input/paper-input';
 import { Component, Prop } from 'vue-property-decorator';
 import { ElementComponent } from '../../components';
 import { mixins } from 'vue-class-component';
Index: build/webpack.dev.conf.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- build/webpack.dev.conf.js	(revision 32dcc44df4a0aa530cbfbbc88981f652e4f660ff)
+++ build/webpack.dev.conf.js	(date 1528888751000)
@@ -23,7 +23,8 @@
 
 module.exports = merge(base.config(TARGET_ENV, BASE_PATH, OUTPUT_PATH), {
   entry: {
-    'test': path.resolve(__dirname, '../test/app/index.ts'),
+    // `babel-polyfill`はIE11対応
+    'test': ['babel-polyfill', path.resolve(__dirname, '../test/app/index.ts')],
   },
 
   plugins: [
Index: src/assets/styles/_colors.styl
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/assets/styles/_colors.styl	(revision 32dcc44df4a0aa530cbfbbc88981f652e4f660ff)
+++ src/assets/styles/_colors.styl	(date 1528891326000)
@@ -1,278 +1,274 @@
-:root {
-
-  /* Material Design color palette from online spec document */
+/* Material Design color palette from online spec document */
 
-  --app-red-50: #ffebee;
-  --app-red-100: #ffcdd2;
-  --app-red-200: #ef9a9a;
-  --app-red-300: #e57373;
-  --app-red-400: #ef5350;
-  --app-red-500: #f44336;
-  --app-red-600: #e53935;
-  --app-red-700: #d32f2f;
-  --app-red-800: #c62828;
-  --app-red-900: #b71c1c;
-  --app-red-a100: #ff8a80;
-  --app-red-a200: #ff5252;
-  --app-red-a400: #ff1744;
-  --app-red-a700: #d50000;
+$app-red-50 = #ffebee;
+$app-red-100 = #ffcdd2;
+$app-red-200 = #ef9a9a;
+$app-red-300 = #e57373;
+$app-red-400 = #ef5350;
+$app-red-500 = #f44336;
+$app-red-600 = #e53935;
+$app-red-700 = #d32f2f;
+$app-red-800 = #c62828;
+$app-red-900 = #b71c1c;
+$app-red-a100 = #ff8a80;
+$app-red-a200 = #ff5252;
+$app-red-a400 = #ff1744;
+$app-red-a700 = #d50000;
 
-  --app-pink-50: #fce4ec;
-  --app-pink-100: #f8bbd0;
-  --app-pink-200: #f48fb1;
-  --app-pink-300: #f06292;
-  --app-pink-400: #ec407a;
-  --app-pink-500: #e91e63;
-  --app-pink-600: #d81b60;
-  --app-pink-700: #c2185b;
-  --app-pink-800: #ad1457;
-  --app-pink-900: #880e4f;
-  --app-pink-a100: #ff80ab;
-  --app-pink-a200: #ff4081;
-  --app-pink-a400: #f50057;
-  --app-pink-a700: #c51162;
+$app-pink-50 = #fce4ec;
+$app-pink-100 = #f8bbd0;
+$app-pink-200 = #f48fb1;
+$app-pink-300 = #f06292;
+$app-pink-400 = #ec407a;
+$app-pink-500 = #e91e63;
+$app-pink-600 = #d81b60;
+$app-pink-700 = #c2185b;
+$app-pink-800 = #ad1457;
+$app-pink-900 = #880e4f;
+$app-pink-a100 = #ff80ab;
+$app-pink-a200 = #ff4081;
+$app-pink-a400 = #f50057;
+$app-pink-a700 = #c51162;
 
-  --app-purple-50: #f3e5f5;
-  --app-purple-100: #e1bee7;
-  --app-purple-200: #ce93d8;
-  --app-purple-300: #ba68c8;
-  --app-purple-400: #ab47bc;
-  --app-purple-500: #9c27b0;
-  --app-purple-600: #8e24aa;
-  --app-purple-700: #7b1fa2;
-  --app-purple-800: #6a1b9a;
-  --app-purple-900: #4a148c;
-  --app-purple-a100: #ea80fc;
-  --app-purple-a200: #e040fb;
-  --app-purple-a400: #d500f9;
-  --app-purple-a700: #aa00ff;
+$app-purple-50 = #f3e5f5;
+$app-purple-100 = #e1bee7;
+$app-purple-200 = #ce93d8;
+$app-purple-300 = #ba68c8;
+$app-purple-400 = #ab47bc;
+$app-purple-500 = #9c27b0;
+$app-purple-600 = #8e24aa;
+$app-purple-700 = #7b1fa2;
+$app-purple-800 = #6a1b9a;
+$app-purple-900 = #4a148c;
+$app-purple-a100 = #ea80fc;
+$app-purple-a200 = #e040fb;
+$app-purple-a400 = #d500f9;
+$app-purple-a700 = #aa00ff;
 
-  --app-deep-purple-50: #ede7f6;
-  --app-deep-purple-100: #d1c4e9;
-  --app-deep-purple-200: #b39ddb;
-  --app-deep-purple-300: #9575cd;
-  --app-deep-purple-400: #7e57c2;
-  --app-deep-purple-500: #673ab7;
-  --app-deep-purple-600: #5e35b1;
-  --app-deep-purple-700: #512da8;
-  --app-deep-purple-800: #4527a0;
-  --app-deep-purple-900: #311b92;
-  --app-deep-purple-a100: #b388ff;
-  --app-deep-purple-a200: #7c4dff;
-  --app-deep-purple-a400: #651fff;
-  --app-deep-purple-a700: #6200ea;
+$app-deep-purple-50 = #ede7f6;
+$app-deep-purple-100 = #d1c4e9;
+$app-deep-purple-200 = #b39ddb;
+$app-deep-purple-300 = #9575cd;
+$app-deep-purple-400 = #7e57c2;
+$app-deep-purple-500 = #673ab7;
+$app-deep-purple-600 = #5e35b1;
+$app-deep-purple-700 = #512da8;
+$app-deep-purple-800 = #4527a0;
+$app-deep-purple-900 = #311b92;
+$app-deep-purple-a100 = #b388ff;
+$app-deep-purple-a200 = #7c4dff;
+$app-deep-purple-a400 = #651fff;
+$app-deep-purple-a700 = #6200ea;
 
-  --app-indigo-50: #e8eaf6;
-  --app-indigo-100: #c5cae9;
-  --app-indigo-200: #9fa8da;
-  --app-indigo-300: #7986cb;
-  --app-indigo-400: #5c6bc0;
-  --app-indigo-500: #3f51b5;
-  --app-indigo-600: #3949ab;
-  --app-indigo-700: #303f9f;
-  --app-indigo-800: #283593;
-  --app-indigo-900: #1a237e;
-  --app-indigo-a100: #8c9eff;
-  --app-indigo-a200: #536dfe;
-  --app-indigo-a400: #3d5afe;
-  --app-indigo-a700: #304ffe;
+$app-indigo-50 = #e8eaf6;
+$app-indigo-100 = #c5cae9;
+$app-indigo-200 = #9fa8da;
+$app-indigo-300 = #7986cb;
+$app-indigo-400 = #5c6bc0;
+$app-indigo-500 = #3f51b5;
+$app-indigo-600 = #3949ab;
+$app-indigo-700 = #303f9f;
+$app-indigo-800 = #283593;
+$app-indigo-900 = #1a237e;
+$app-indigo-a100 = #8c9eff;
+$app-indigo-a200 = #536dfe;
+$app-indigo-a400 = #3d5afe;
+$app-indigo-a700 = #304ffe;
 
-  --app-blue-50: #e3f2fd;
-  --app-blue-100: #bbdefb;
-  --app-blue-200: #90caf9;
-  --app-blue-300: #64b5f6;
-  --app-blue-400: #42a5f5;
-  --app-blue-500: #2196f3;
-  --app-blue-600: #1e88e5;
-  --app-blue-700: #1976d2;
-  --app-blue-800: #1565c0;
-  --app-blue-900: #0d47a1;
-  --app-blue-a100: #82b1ff;
-  --app-blue-a200: #448aff;
-  --app-blue-a400: #2979ff;
-  --app-blue-a700: #2962ff;
+$app-blue-50 = #e3f2fd;
+$app-blue-100 = #bbdefb;
+$app-blue-200 = #90caf9;
+$app-blue-300 = #64b5f6;
+$app-blue-400 = #42a5f5;
+$app-blue-500 = #2196f3;
+$app-blue-600 = #1e88e5;
+$app-blue-700 = #1976d2;
+$app-blue-800 = #1565c0;
+$app-blue-900 = #0d47a1;
+$app-blue-a100 = #82b1ff;
+$app-blue-a200 = #448aff;
+$app-blue-a400 = #2979ff;
+$app-blue-a700 = #2962ff;
 
-  --app-light-blue-50: #e1f5fe;
-  --app-light-blue-100: #b3e5fc;
-  --app-light-blue-200: #81d4fa;
-  --app-light-blue-300: #4fc3f7;
-  --app-light-blue-400: #29b6f6;
-  --app-light-blue-500: #03a9f4;
-  --app-light-blue-600: #039be5;
-  --app-light-blue-700: #0288d1;
-  --app-light-blue-800: #0277bd;
-  --app-light-blue-900: #01579b;
-  --app-light-blue-a100: #80d8ff;
-  --app-light-blue-a200: #40c4ff;
-  --app-light-blue-a400: #00b0ff;
-  --app-light-blue-a700: #0091ea;
+$app-light-blue-50 = #e1f5fe;
+$app-light-blue-100 = #b3e5fc;
+$app-light-blue-200 = #81d4fa;
+$app-light-blue-300 = #4fc3f7;
+$app-light-blue-400 = #29b6f6;
+$app-light-blue-500 = #03a9f4;
+$app-light-blue-600 = #039be5;
+$app-light-blue-700 = #0288d1;
+$app-light-blue-800 = #0277bd;
+$app-light-blue-900 = #01579b;
+$app-light-blue-a100 = #80d8ff;
+$app-light-blue-a200 = #40c4ff;
+$app-light-blue-a400 = #00b0ff;
+$app-light-blue-a700 = #0091ea;
 
-  --app-cyan-50: #e0f7fa;
-  --app-cyan-100: #b2ebf2;
-  --app-cyan-200: #80deea;
-  --app-cyan-300: #4dd0e1;
-  --app-cyan-400: #26c6da;
-  --app-cyan-500: #00bcd4;
-  --app-cyan-600: #00acc1;
-  --app-cyan-700: #0097a7;
-  --app-cyan-800: #00838f;
-  --app-cyan-900: #006064;
-  --app-cyan-a100: #84ffff;
-  --app-cyan-a200: #18ffff;
-  --app-cyan-a400: #00e5ff;
-  --app-cyan-a700: #00b8d4;
+$app-cyan-50 = #e0f7fa;
+$app-cyan-100 = #b2ebf2;
+$app-cyan-200 = #80deea;
+$app-cyan-300 = #4dd0e1;
+$app-cyan-400 = #26c6da;
+$app-cyan-500 = #00bcd4;
+$app-cyan-600 = #00acc1;
+$app-cyan-700 = #0097a7;
+$app-cyan-800 = #00838f;
+$app-cyan-900 = #006064;
+$app-cyan-a100 = #84ffff;
+$app-cyan-a200 = #18ffff;
+$app-cyan-a400 = #00e5ff;
+$app-cyan-a700 = #00b8d4;
 
-  --app-teal-50: #e0f2f1;
-  --app-teal-100: #b2dfdb;
-  --app-teal-200: #80cbc4;
-  --app-teal-300: #4db6ac;
-  --app-teal-400: #26a69a;
-  --app-teal-500: #009688;
-  --app-teal-600: #00897b;
-  --app-teal-700: #00796b;
-  --app-teal-800: #00695c;
-  --app-teal-900: #004d40;
-  --app-teal-a100: #a7ffeb;
-  --app-teal-a200: #64ffda;
-  --app-teal-a400: #1de9b6;
-  --app-teal-a700: #00bfa5;
+$app-teal-50 = #e0f2f1;
+$app-teal-100 = #b2dfdb;
+$app-teal-200 = #80cbc4;
+$app-teal-300 = #4db6ac;
+$app-teal-400 = #26a69a;
+$app-teal-500 = #009688;
+$app-teal-600 = #00897b;
+$app-teal-700 = #00796b;
+$app-teal-800 = #00695c;
+$app-teal-900 = #004d40;
+$app-teal-a100 = #a7ffeb;
+$app-teal-a200 = #64ffda;
+$app-teal-a400 = #1de9b6;
+$app-teal-a700 = #00bfa5;
 
-  --app-green-50: #e8f5e9;
-  --app-green-100: #c8e6c9;
-  --app-green-200: #a5d6a7;
-  --app-green-300: #81c784;
-  --app-green-400: #66bb6a;
-  --app-green-500: #4caf50;
-  --app-green-600: #43a047;
-  --app-green-700: #388e3c;
-  --app-green-800: #2e7d32;
-  --app-green-900: #1b5e20;
-  --app-green-a100: #b9f6ca;
-  --app-green-a200: #69f0ae;
-  --app-green-a400: #00e676;
-  --app-green-a700: #00c853;
+$app-green-50 = #e8f5e9;
+$app-green-100 = #c8e6c9;
+$app-green-200 = #a5d6a7;
+$app-green-300 = #81c784;
+$app-green-400 = #66bb6a;
+$app-green-500 = #4caf50;
+$app-green-600 = #43a047;
+$app-green-700 = #388e3c;
+$app-green-800 = #2e7d32;
+$app-green-900 = #1b5e20;
+$app-green-a100 = #b9f6ca;
+$app-green-a200 = #69f0ae;
+$app-green-a400 = #00e676;
+$app-green-a700 = #00c853;
 
-  --app-light-green-50: #f1f8e9;
-  --app-light-green-100: #dcedc8;
-  --app-light-green-200: #c5e1a5;
-  --app-light-green-300: #aed581;
-  --app-light-green-400: #9ccc65;
-  --app-light-green-500: #8bc34a;
-  --app-light-green-600: #7cb342;
-  --app-light-green-700: #689f38;
-  --app-light-green-800: #558b2f;
-  --app-light-green-900: #33691e;
-  --app-light-green-a100: #ccff90;
-  --app-light-green-a200: #b2ff59;
-  --app-light-green-a400: #76ff03;
-  --app-light-green-a700: #64dd17;
+$app-light-green-50 = #f1f8e9;
+$app-light-green-100 = #dcedc8;
+$app-light-green-200 = #c5e1a5;
+$app-light-green-300 = #aed581;
+$app-light-green-400 = #9ccc65;
+$app-light-green-500 = #8bc34a;
+$app-light-green-600 = #7cb342;
+$app-light-green-700 = #689f38;
+$app-light-green-800 = #558b2f;
+$app-light-green-900 = #33691e;
+$app-light-green-a100 = #ccff90;
+$app-light-green-a200 = #b2ff59;
+$app-light-green-a400 = #76ff03;
+$app-light-green-a700 = #64dd17;
 
-  --app-lime-50: #f9fbe7;
-  --app-lime-100: #f0f4c3;
-  --app-lime-200: #e6ee9c;
-  --app-lime-300: #dce775;
-  --app-lime-400: #d4e157;
-  --app-lime-500: #cddc39;
-  --app-lime-600: #c0ca33;
-  --app-lime-700: #afb42b;
-  --app-lime-800: #9e9d24;
-  --app-lime-900: #827717;
-  --app-lime-a100: #f4ff81;
-  --app-lime-a200: #eeff41;
-  --app-lime-a400: #c6ff00;
-  --app-lime-a700: #aeea00;
+$app-lime-50 = #f9fbe7;
+$app-lime-100 = #f0f4c3;
+$app-lime-200 = #e6ee9c;
+$app-lime-300 = #dce775;
+$app-lime-400 = #d4e157;
+$app-lime-500 = #cddc39;
+$app-lime-600 = #c0ca33;
+$app-lime-700 = #afb42b;
+$app-lime-800 = #9e9d24;
+$app-lime-900 = #827717;
+$app-lime-a100 = #f4ff81;
+$app-lime-a200 = #eeff41;
+$app-lime-a400 = #c6ff00;
+$app-lime-a700 = #aeea00;
 
-  --app-yellow-50: #fffde7;
-  --app-yellow-100: #fff9c4;
-  --app-yellow-200: #fff59d;
-  --app-yellow-300: #fff176;
-  --app-yellow-400: #ffee58;
-  --app-yellow-500: #ffeb3b;
-  --app-yellow-600: #fdd835;
-  --app-yellow-700: #fbc02d;
-  --app-yellow-800: #f9a825;
-  --app-yellow-900: #f57f17;
-  --app-yellow-a100: #ffff8d;
-  --app-yellow-a200: #ffff00;
-  --app-yellow-a400: #ffea00;
-  --app-yellow-a700: #ffd600;
+$app-yellow-50 = #fffde7;
+$app-yellow-100 = #fff9c4;
+$app-yellow-200 = #fff59d;
+$app-yellow-300 = #fff176;
+$app-yellow-400 = #ffee58;
+$app-yellow-500 = #ffeb3b;
+$app-yellow-600 = #fdd835;
+$app-yellow-700 = #fbc02d;
+$app-yellow-800 = #f9a825;
+$app-yellow-900 = #f57f17;
+$app-yellow-a100 = #ffff8d;
+$app-yellow-a200 = #ffff00;
+$app-yellow-a400 = #ffea00;
+$app-yellow-a700 = #ffd600;
 
-  --app-amber-50: #fff8e1;
-  --app-amber-100: #ffecb3;
-  --app-amber-200: #ffe082;
-  --app-amber-300: #ffd54f;
-  --app-amber-400: #ffca28;
-  --app-amber-500: #ffc107;
-  --app-amber-600: #ffb300;
-  --app-amber-700: #ffa000;
-  --app-amber-800: #ff8f00;
-  --app-amber-900: #ff6f00;
-  --app-amber-a100: #ffe57f;
-  --app-amber-a200: #ffd740;
-  --app-amber-a400: #ffc400;
-  --app-amber-a700: #ffab00;
+$app-amber-50 = #fff8e1;
+$app-amber-100 = #ffecb3;
+$app-amber-200 = #ffe082;
+$app-amber-300 = #ffd54f;
+$app-amber-400 = #ffca28;
+$app-amber-500 = #ffc107;
+$app-amber-600 = #ffb300;
+$app-amber-700 = #ffa000;
+$app-amber-800 = #ff8f00;
+$app-amber-900 = #ff6f00;
+$app-amber-a100 = #ffe57f;
+$app-amber-a200 = #ffd740;
+$app-amber-a400 = #ffc400;
+$app-amber-a700 = #ffab00;
 
-  --app-orange-50: #fff3e0;
-  --app-orange-100: #ffe0b2;
-  --app-orange-200: #ffcc80;
-  --app-orange-300: #ffb74d;
-  --app-orange-400: #ffa726;
-  --app-orange-500: #ff9800;
-  --app-orange-600: #fb8c00;
-  --app-orange-700: #f57c00;
-  --app-orange-800: #ef6c00;
-  --app-orange-900: #e65100;
-  --app-orange-a100: #ffd180;
-  --app-orange-a200: #ffab40;
-  --app-orange-a400: #ff9100;
-  --app-orange-a700: #ff6500;
+$app-orange-50 = #fff3e0;
+$app-orange-100 = #ffe0b2;
+$app-orange-200 = #ffcc80;
+$app-orange-300 = #ffb74d;
+$app-orange-400 = #ffa726;
+$app-orange-500 = #ff9800;
+$app-orange-600 = #fb8c00;
+$app-orange-700 = #f57c00;
+$app-orange-800 = #ef6c00;
+$app-orange-900 = #e65100;
+$app-orange-a100 = #ffd180;
+$app-orange-a200 = #ffab40;
+$app-orange-a400 = #ff9100;
+$app-orange-a700 = #ff6500;
 
-  --app-deep-orange-50: #fbe9e7;
-  --app-deep-orange-100: #ffccbc;
-  --app-deep-orange-200: #ffab91;
-  --app-deep-orange-300: #ff8a65;
-  --app-deep-orange-400: #ff7043;
-  --app-deep-orange-500: #ff5722;
-  --app-deep-orange-600: #f4511e;
-  --app-deep-orange-700: #e64a19;
-  --app-deep-orange-800: #d84315;
-  --app-deep-orange-900: #bf360c;
-  --app-deep-orange-a100: #ff9e80;
-  --app-deep-orange-a200: #ff6e40;
-  --app-deep-orange-a400: #ff3d00;
-  --app-deep-orange-a700: #dd2c00;
+$app-deep-orange-50 = #fbe9e7;
+$app-deep-orange-100 = #ffccbc;
+$app-deep-orange-200 = #ffab91;
+$app-deep-orange-300 = #ff8a65;
+$app-deep-orange-400 = #ff7043;
+$app-deep-orange-500 = #ff5722;
+$app-deep-orange-600 = #f4511e;
+$app-deep-orange-700 = #e64a19;
+$app-deep-orange-800 = #d84315;
+$app-deep-orange-900 = #bf360c;
+$app-deep-orange-a100 = #ff9e80;
+$app-deep-orange-a200 = #ff6e40;
+$app-deep-orange-a400 = #ff3d00;
+$app-deep-orange-a700 = #dd2c00;
 
-  --app-brown-50: #efebe9;
-  --app-brown-100: #d7ccc8;
-  --app-brown-200: #bcaaa4;
-  --app-brown-300: #a1887f;
-  --app-brown-400: #8d6e63;
-  --app-brown-500: #795548;
-  --app-brown-600: #6d4c41;
-  --app-brown-700: #5d4037;
-  --app-brown-800: #4e342e;
-  --app-brown-900: #3e2723;
+$app-brown-50 = #efebe9;
+$app-brown-100 = #d7ccc8;
+$app-brown-200 = #bcaaa4;
+$app-brown-300 = #a1887f;
+$app-brown-400 = #8d6e63;
+$app-brown-500 = #795548;
+$app-brown-600 = #6d4c41;
+$app-brown-700 = #5d4037;
+$app-brown-800 = #4e342e;
+$app-brown-900 = #3e2723;
 
-  --app-grey-50: #fafafa;
-  --app-grey-100: #f5f5f5;
-  --app-grey-200: #eeeeee;
-  --app-grey-300: #e0e0e0;
-  --app-grey-400: #bdbdbd;
-  --app-grey-500: #9e9e9e;
-  --app-grey-600: #757575;
-  --app-grey-700: #616161;
-  --app-grey-800: #424242;
-  --app-grey-900: #212121;
+$app-grey-50 = #fafafa;
+$app-grey-100 = #f5f5f5;
+$app-grey-200 = #eeeeee;
+$app-grey-300 = #e0e0e0;
+$app-grey-400 = #bdbdbd;
+$app-grey-500 = #9e9e9e;
+$app-grey-600 = #757575;
+$app-grey-700 = #616161;
+$app-grey-800 = #424242;
+$app-grey-900 = #212121;
 
-  --app-blue-grey-50: #eceff1;
-  --app-blue-grey-100: #cfd8dc;
-  --app-blue-grey-200: #b0bec5;
-  --app-blue-grey-300: #90a4ae;
-  --app-blue-grey-400: #78909c;
-  --app-blue-grey-500: #607d8b;
-  --app-blue-grey-600: #546e7a;
-  --app-blue-grey-700: #455a64;
-  --app-blue-grey-800: #37474f;
-  --app-blue-grey-900: #263238;
-
-}
+$app-blue-grey-50 = #eceff1;
+$app-blue-grey-100 = #cfd8dc;
+$app-blue-grey-200 = #b0bec5;
+$app-blue-grey-300 = #90a4ae;
+$app-blue-grey-400 = #78909c;
+$app-blue-grey-500 = #607d8b;
+$app-blue-grey-600 = #546e7a;
+$app-blue-grey-700 = #455a64;
+$app-blue-grey-800 = #37474f;
+$app-blue-grey-900 = #263238;
Index: src/app/views/shopping-view/index.vue
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/app/views/shopping-view/index.vue	(revision 32dcc44df4a0aa530cbfbbc88981f652e4f660ff)
+++ src/app/views/shopping-view/index.vue	(date 1528892179000)
@@ -1,6 +1,5 @@
 <style lang="stylus" scoped>
-  @import '../../../assets/styles/_spacing.styl';
-  @import '../../../assets/styles/_typography.styl';
+  @import '../../../assets/styles/_variables.styl';
 
   .title-text {
     @extend .app-font-title;
@@ -10,7 +9,7 @@
     @extend .app-pa-3;
 
     &.iron-selected {
-      background-color: var(--app-grid-selected-activ-item);
+      background-color: $app-grid-selected-activ-item;
     }
 
     .title {
@@ -19,17 +18,13 @@
 
     .detail {
       @extend .app-font-body1;
-      color: var(--app-secondary-text-color);
+      color: $app-secondary-text-color;
     }
   }
 
   .error-text {
     @extend .app-font-body1;
-    color: var(--app-error-text-color);
-  }
-
-  .checkout-button {
-    color: var(--app-link-color);
+    color: $app-error-text-color;
   }
 </style>
 
@@ -54,7 +49,7 @@
           </div>
         </div>
         <div class="flex"></div>
-        <paper-icon-button icon="icons:add-box" @click="addProductToCart(product)"></paper-icon-button>
+        <button @click="addProductToCart(product)">＋</button>
       </div>
     </div>
 
@@ -77,11 +72,10 @@
       </div>
       <div class="layout horizontal center">
         <div class="flex error-text">{{ checkoutStatus.message }}</div>
-        <paper-button
+        <button
           v-show="!cartIsEmpty"
-          class="checkout-button"
           @click="checkout">Checkout
-        </paper-button>
+        </button>
       </div>
     </div>
 
@@ -90,9 +84,6 @@
 
 
 <script lang="ts">
-import '@polymer/paper-button/paper-button';
-import '@polymer/paper-card/paper-card';
-import '@polymer/paper-icon-button/paper-icon-button';
 import { CartProduct, CheckoutStatus, Product } from '../../stores/types';
 import { Component } from 'vue-property-decorator';
 import { ElementComponent } from '../../components';
Index: src/app/index.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/app/index.ts	(revision 32dcc44df4a0aa530cbfbbc88981f652e4f660ff)
+++ src/app/index.ts	(date 1528891326000)
@@ -1,4 +1,3 @@
-import '../assets/styles/custom-style';
 import '../assets/styles/main.styl';
 
 import * as apis from './apis';
