Index: src/store/counter-module/index.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/store/counter-module/index.ts	(date 1548951392000)
+++ src/store/counter-module/index.ts	(date 1548951392000)
@@ -0,0 +1,28 @@
+import { GetterTree, MutationTree, ActionTree } from 'vuex'
+import { CounterModule, CounterState, CounterTypes, RootState } from '../types'
+
+export class CounterModuleImpl implements CounterModule {
+  namespaced = true
+
+  state: CounterState = {
+    counter: 1,
+  }
+
+  getters: GetterTree<CounterState, RootState> = {
+    [CounterTypes.CURRENT](state: CounterState): number {
+      return state.counter
+    },
+  }
+
+  mutations: MutationTree<CounterState> = {
+    [CounterTypes.INCREMENT](state: CounterState): void {
+      state.counter++
+    },
+  }
+
+  actions: ActionTree<CounterState, RootState> = {}
+}
+
+export function newCounterModule() {
+  return new CounterModuleImpl()
+}
Index: src/store/app-store.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/store/app-store.ts	(date 1548951392000)
+++ src/store/app-store.ts	(date 1548951392000)
@@ -0,0 +1,121 @@
+import Vue from 'vue'
+import { CartItem, CartTypes, CheckoutStatus, CounterTypes, Product, ProductTypes, RootState } from '@/store/types'
+import { Component } from 'vue-property-decorator'
+import { Store } from 'vuex'
+
+@Component
+export class AppStore extends Vue {
+  m_store: Store<RootState>
+
+  setStore(store: Store<RootState>) {
+    this.counter.setStore(store)
+    this.product.setStore(store)
+    this.cart.setStore(store)
+  }
+
+  m_counter = new CounterModule()
+
+  get counter(): CounterModule {
+    return this.m_counter
+  }
+
+  m_product = new ProductModule()
+
+  get product(): ProductModule {
+    return this.m_product
+  }
+
+  m_cart = new CartModule()
+
+  get cart(): CartModule {
+    return this.m_cart
+  }
+}
+
+@Component
+class CounterModule extends Vue {
+
+  m_store: Store<RootState>
+
+  setStore(store: Store<RootState>) {
+    this.m_store = store
+  }
+
+  get current(): number {
+    return this.m_store.getters[`${CounterTypes.PATH}/${CounterTypes.CURRENT}`]
+  }
+
+  increment(): void {
+    this.m_store.commit(`${CounterTypes.PATH}/${CounterTypes.INCREMENT}`)
+  }
+}
+
+@Component
+class ProductModule extends Vue {
+
+  m_store: Store<RootState>
+
+  setStore(store: Store<RootState>) {
+    this.m_store = store
+  }
+
+  get allProducts(): Product[] {
+    return this.m_store.getters[`${ProductTypes.PATH}/${ProductTypes.ALL_PRODUCTS}`]
+  }
+
+  getProductById(productId: string): Product | undefined {
+    return this.m_store.getters[`${ProductTypes.PATH}/${ProductTypes.GET_PRODUCT_BY_ID}`](productId)
+  }
+
+  pullAllProducts(): Promise<void> {
+    return this.m_store.dispatch(`${ProductTypes.PATH}/${ProductTypes.PULL_ALL_PRODUCTS}`)
+  }
+
+  setProducts(products: Product[]): void {
+    return this.m_store.commit(`${ProductTypes.PATH}/${ProductTypes.SET_PRODUCTS}`, products)
+  }
+
+  decrementProductInventory(productId: string): void {
+    return this.m_store.commit(`${ProductTypes.PATH}/${ProductTypes.DECREMENT_PRODUCT_INVENTORY}`, productId)
+  }
+}
+
+@Component
+class CartModule extends Vue {
+
+  m_store: Store<RootState>
+
+  setStore(store: Store<RootState>) {
+    this.m_store = store
+  }
+
+  get checkoutStatus(): CheckoutStatus {
+    return this.m_store.getters[`${CartTypes.PATH}/${CartTypes.CHECKOUT_STATUS}`]
+  }
+
+  get cartItems(): CartItem[] {
+    return this.m_store.getters[`${CartTypes.PATH}/${CartTypes.CART_ITEMS}`]
+  }
+
+  get cartTotalPrice(): number {
+    return this.m_store.getters[`${CartTypes.PATH}/${CartTypes.CART_TOTAL_PRICE}`]
+  }
+
+  getCartItemById(productId: string): Product | undefined {
+    return this.m_store.getters[`${CartTypes.PATH}/${CartTypes.GET_CART_ITEM_BY_ID}`](productId)
+  }
+
+  checkout(): Promise<void> {
+    return this.m_store.dispatch(`${CartTypes.PATH}/${CartTypes.CHECKOUT}`)
+  }
+
+  addProductToCart(productId: string): Promise<void> {
+    return this.m_store.dispatch(`${CartTypes.PATH}/${CartTypes.ADD_PRODUCT_TO_CART}`, productId)
+  }
+}
+
+export function newAppStore(store: Store<RootState>) {
+  const result = new AppStore()
+  result.setStore(store)
+  return result
+}
Index: src/store/base.ts
===================================================================
--- src/store/base.ts	(date 1548949595000)
+++ src/store/base.ts	(date 1548949595000)
@@ -1,32 +0,0 @@
-import Vue from 'vue'
-import { APIs } from '@/apis'
-
-export abstract class BaseModule<S> extends Vue {
-  //----------------------------------------------------------------------
-  //
-  //  Variables
-  //
-  //----------------------------------------------------------------------
-
-  protected readonly $apis: APIs
-
-  m_state: S
-
-  get f_state(): S {
-    return this.m_state
-  }
-
-  //----------------------------------------------------------------------
-  //
-  //  Internal methods
-  //
-  //----------------------------------------------------------------------
-
-  /**
-   * Moduleにひも付くStateを初期化します。
-   * @param state
-   */
-  f_initState(state: S): void {
-    this.m_state = state
-  }
-}
Index: test/api/shop-api/index.ts
===================================================================
--- test/api/shop-api/index.ts	(date 1548949595000)
+++ test/api/shop-api/index.ts	(date 1548949595000)
@@ -1,9 +0,0 @@
-import _shopAPI from '@/apis/shop-api'
-
-const assert = chai.assert
-
-suite('api/shop-api', () => {
-  test('サンプルテスト', () => {
-    assert.equal(1, 1)
-  })
-})
Index: test/store/cart-module/index.ts
===================================================================
--- test/store/cart-module/index.ts	(date 1548949595000)
+++ test/store/cart-module/index.ts	(date 1548949595000)
@@ -1,109 +0,0 @@
-import * as td from 'testdouble'
-import { newProductModule } from '@/store/product-module'
-import { CartState, CartModuleImpl, newCartModule } from '@/store/cart-module'
-import { CheckoutStatus } from '@/store'
-import { Product as APIProduct } from '@/apis'
-import { TestModule } from '../../types'
-
-const assert = chai.assert
-
-suite('store/cart-module', () => {
-  const productModule = newProductModule()
-  const cartModule = newCartModule({ product: productModule }) as CartModuleImpl & TestModule<CartState>
-  const shopAPI = cartModule.$apis.shop
-
-  const PRODUCTS: APIProduct[] = [
-    { id: '1', title: 'iPad 4 Mini', price: 500.01, inventory: 2 },
-    { id: '2', title: 'H&M T-Shirt White', price: 10.99, inventory: 10 },
-    { id: '3', title: 'Charli XCX - Sucker CD', price: 19.99, inventory: 5 },
-  ]
-
-  setup(() => {
-    cartModule.f_initState({
-      items: [],
-      checkoutStatus: CheckoutStatus.None,
-    })
-  })
-
-  teardown(() => {
-    cartModule.f_initState({
-      items: [],
-      checkoutStatus: CheckoutStatus.None,
-    })
-    td.reset()
-  })
-
-  test('getCartItemById() - 一般ケース', () => {
-    cartModule.f_state.items = [ { id: '1', quantity: 1 } ]
-    const product = PRODUCTS[0]
-    td.replace(cartModule, 'm_getProductById')
-    td.when(cartModule.m_getProductById(product.id)).thenReturn(product)
-
-    const actual = cartModule.getCartItemById(product.id)
-    assert.deepEqual(actual, {
-      id: product.id,
-      title: product.title,
-      price: product.price,
-      quantity: 1,
-    })
-  })
-
-  test('getCartItemById() - 存在しない商品IDを指定した場合', () => {
-    assert.throws(() => cartModule.getCartItemById('9876'), Error, 'A product that matches the specified productId "9876" was not found.')
-  })
-
-  test('addProductToCart() - 一般ケース', () => {
-    const product = PRODUCTS[1]
-    // 【準備】
-    td.replace(cartModule, 'm_getProductById')
-    td.when(cartModule.m_getProductById(product.id)).thenReturn(product)
-
-    const decrementProductInventory = td.replace(productModule, 'decrementProductInventory')
-
-    // 【実行】
-    // `addProductToCart()`を2回実行
-    cartModule.addProductToCart(product.id)
-    cartModule.addProductToCart(product.id)
-
-    // 【検証】
-    assert.equal(cartModule.f_state.checkoutStatus, CheckoutStatus.None)
-    // カートに追加された商品とその数量を検証
-    const cartItem = cartModule.getCartItemById(product.id)
-    assert.equal(cartItem!.id, product.id)
-    assert.equal(cartItem!.quantity, 2)
-
-    // `ProductModule#decrementProductInventory()`の呼び出し回数と渡された引数を検証
-    const decrementProductInventoryExplain = td.explain(decrementProductInventory)
-    assert.equal(decrementProductInventoryExplain.callCount, 2)
-    assert.equal(decrementProductInventoryExplain.calls[0].args[0], product.id)
-    assert.equal(decrementProductInventoryExplain.calls[1].args[0], product.id)
-  })
-
-  test('checkout() - 一般ケース', async () => {
-    const CART_ITEMS = [ { id: '1', quantity: 1 }, { id: '2', quantity: 1 } ]
-    cartModule.f_state.items = CART_ITEMS
-    const buyProducts = td.replace(shopAPI, 'buyProducts')
-    td.when(shopAPI.buyProducts(CART_ITEMS)).thenResolve()
-
-    await cartModule.checkout()
-    assert.equal(cartModule.f_state.checkoutStatus, CheckoutStatus.Successful)
-    assert.deepEqual(cartModule.f_state.items, [])
-
-    // `ShopAPI#buyProducts()`の呼び出し回数と渡された引数を検証
-    const buyProductsExplain = td.explain(buyProducts)
-    assert.equal(buyProductsExplain.callCount, 1)
-    assert.deepEqual(buyProductsExplain.calls[0].args[0], CART_ITEMS)
-  })
-
-  test('checkout() - エラーケース', async () => {
-    const CART_ITEMS = [ { id: '1', quantity: 1 }, { id: '2', quantity: 1 } ]
-    cartModule.f_state.items = CART_ITEMS
-
-    const buyProducts = td.replace(shopAPI, 'buyProducts')
-    td.when(shopAPI.buyProducts(CART_ITEMS)).thenReject(new Error())
-
-    await cartModule.checkout()
-    assert.equal(cartModule.f_state.checkoutStatus, CheckoutStatus.Failed)
-    assert.deepEqual(cartModule.f_state.items, CART_ITEMS)
-  })
-})
Index: test/store/product-module/index.ts
===================================================================
--- test/store/product-module/index.ts	(date 1548949595000)
+++ test/store/product-module/index.ts	(date 1548949595000)
@@ -1,64 +0,0 @@
-import * as td from 'testdouble'
-import { Product as APIProduct } from '@/apis'
-import { Product } from '@/store'
-import { TestModule } from '../../types'
-import { newProductModule, ProductState, ProductModuleImpl } from '@/store/product-module'
-
-const assert = chai.assert
-
-suite('store/product-module', () => {
-  const productModule = newProductModule() as ProductModuleImpl & TestModule<ProductState>
-  const shopAPI = productModule.$apis.shop
-
-  const API_PRODUCTS: APIProduct[] = [
-    { id: '1', title: 'iPad 4 Mini', price: 500.01, inventory: 2 },
-    { id: '2', title: 'H&M T-Shirt White', price: 10.99, inventory: 10 },
-    { id: '3', title: 'Charli XCX - Sucker CD', price: 19.99, inventory: 5 },
-  ]
-
-  setup(() => {
-    productModule.f_initState({
-      all: API_PRODUCTS,
-    })
-  })
-
-  teardown(() => {
-    td.reset()
-  })
-
-  test('getProductById() - 取得できるパターン', () => {
-    const stateProduct = productModule.m_getStateProductById('1') as Product
-    const actual = productModule.getProductById(stateProduct.id)
-    assert.deepEqual(actual, stateProduct)
-    // actualとproductが同一オブジェクトでないことを検証
-    // (つまりコピーであることを検証)
-    assert.notEqual(actual, stateProduct)
-  })
-
-  test('getProductById() - 取得できないパターン', () => {
-    const actual = productModule.getProductById('9876')
-    assert.isUndefined(actual)
-  })
-
-  test('decrementProductInventory() - 一般ケース', () => {
-    const stateProduct = productModule.m_getStateProductById('1') as Product
-    const inventoryBk = stateProduct.inventory
-    productModule.decrementProductInventory(stateProduct.id)
-    assert.equal(stateProduct.inventory, inventoryBk - 1)
-  })
-
-  test('decrementProductInventory() - 存在しない商品IDを指定した場合', () => {
-    productModule.decrementProductInventory('9876')
-    // 何も問題は起きない
-    assert(true)
-  })
-
-  test('pullAllProducts()', async () => {
-    const NEW_API_PRODUCTS = [ { id: '1', title: 'product1', price: 101, inventory: 1 }, { id: '2', title: 'product2', price: 102, inventory: 2 } ]
-    td.replace(shopAPI, 'getProducts')
-    td.when(shopAPI.getProducts()).thenResolve(NEW_API_PRODUCTS)
-
-    await productModule.pullAllProducts()
-    assert.deepEqual(productModule.allProducts, NEW_API_PRODUCTS)
-  })
-})
Index: test/types/index.d.ts
===================================================================
--- test/types/index.d.ts	(date 1548949595000)
+++ test/types/index.d.ts	(date 1548949595000)
@@ -1,7 +0,0 @@
-import { APIs } from '@/apis'
-import { AppStore } from '@/store'
-
-declare interface TestModule<S> {
-  $apis: APIs
-  $appStore: AppStore
-}
Index: test/test.html
===================================================================
--- test/test.html	(date 1548949595000)
+++ test/test.html	(date 1548949595000)
@@ -1,19 +0,0 @@
-<!DOCTYPE html>
-<html lang="en">
-  <head>
-    <meta charset="utf-8" />
-    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
-    <meta name="viewport" content="width=device-width,initial-scale=1.0" />
-    <base href="<%= BASE_URL %>" />
-    <link rel="icon" href="<%= BASE_URL %>favicon.ico" />
-    <title><%= htmlWebpackPlugin.options.title %></title>
-
-    <link href="node_modules/mocha/mocha.css" rel="stylesheet" />
-    <script src="node_modules/mocha/mocha.js"></script>
-    <script src="node_modules/chai/chai.js"></script>
-  </head>
-  <body>
-    <div id="mocha"></div>
-    <!-- built files will be auto injected -->
-  </body>
-</html>
Index: test/test.ts
===================================================================
--- test/test.ts	(date 1548949595000)
+++ test/test.ts	(date 1548949595000)
@@ -1,17 +0,0 @@
-(async () => {
-  mocha.setup({ ui: 'tdd' })
-
-  const utils = await import('../src/base/utils')
-  const config = await import('../src/base/config')
-  const apis = await import('../src/apis')
-  const store = await import('../src/store')
-
-  utils.initUtils()
-  config.initConfig()
-  apis.initAPI()
-  store.initStore()
-
-  await Promise.all([ import('./api/shop-api'), import('./store/cart-module'), import('./store/product-module') ])
-
-  mocha.run()
-})()
Index: src/store/product-module/index.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/store/product-module/index.ts	(date 1548949595000)
+++ src/store/product-module/index.ts	(date 1548951392000)
@@ -1,77 +1,45 @@
-import { BaseModule } from '@/store/base'
-import { Component } from 'vue-property-decorator'
-import { NoCache } from '@/base/component'
-import { Product, ProductModule } from '@/store/types'
+import { GetterTree, MutationTree, ActionTree } from 'vuex'
+import { ProductModule, ProductState, RootState, Product, ProductTypes } from '../types'
+import { apis } from '@/apis'
 
-export interface ProductState {
-  all: Product[]
-}
+export class ProductModuleImpl implements ProductModule {
+  namespaced = true
 
-@Component
-export class ProductModuleImpl extends BaseModule<ProductState> implements ProductModule {
-  //----------------------------------------------------------------------
-  //
-  //  Constructors
-  //
-  //----------------------------------------------------------------------
-
-  constructor() {
-    super()
-    this.f_initState({
-      all: [],
-    })
+  state: ProductState = {
+    all: [],
   }
 
-  //----------------------------------------------------------------------
-  //
-  //  Properties
-  //
-  //----------------------------------------------------------------------
+  getters: GetterTree<ProductState, RootState> = {
+    [ProductTypes.ALL_PRODUCTS](state): Product[] {
+      return state.all
+    },
 
-  @NoCache
-  get allProducts(): Product[] {
-    return this.$utils.cloneDeep(this.f_state.all)
-  }
-
-  //----------------------------------------------------------------------
-  //
-  //  Lifecycle hooks
-  //
-  //----------------------------------------------------------------------
-
-  created() {}
-
-  //----------------------------------------------------------------------
-  //
-  //  Methods
-  //
-  //----------------------------------------------------------------------
-
-  getProductById(productId: string): Product | undefined {
-    const stateProduct = this.m_getStateProductById(productId)
-    return this.$utils.cloneDeep(stateProduct)
+    [ProductTypes.GET_PRODUCT_BY_ID](state) {
+      return (productId: string) => {
+        const product = state.all.find(item => item.id === productId)
+        return product
+      }
+    },
   }
 
-  decrementProductInventory(productId: string): void {
-    const stateProduct = this.m_getStateProductById(productId)
-    if (stateProduct) {
-      stateProduct.inventory--
-    }
-  }
+  mutations: MutationTree<ProductState> = {
+    [ProductTypes.SET_PRODUCTS](state, products: Product[]): void {
+      state.all = products
+    },
 
-  async pullAllProducts(): Promise<void> {
-    const products = await this.$apis.shop.getProducts()
-    this.f_state.all = products
+    [ProductTypes.DECREMENT_PRODUCT_INVENTORY](state, productId: string): void {
+      const product = state.all.find(item => item.id === productId)
+      if (product) {
+        product.inventory--
+      }
+    },
   }
-
-  //----------------------------------------------------------------------
-  //
-  //  Internal methods
-  //
-  //----------------------------------------------------------------------
 
-  m_getStateProductById(productId: string): Product | undefined {
-    return this.f_state.all.find((item) => item.id === productId)
+  actions: ActionTree<ProductState, RootState> = {
+    async [ProductTypes.PULL_ALL_PRODUCTS](context): Promise<void> {
+      const products = await apis.shop.getProducts()
+      context.commit(ProductTypes.SET_PRODUCTS, products)
+    },
   }
 }
 
Index: src/views/shopping-view/index.vue
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/views/shopping-view/index.vue	(date 1548949595000)
+++ src/views/shopping-view/index.vue	(date 1548951392000)
@@ -118,8 +118,8 @@
   //
   //----------------------------------------------------------------------
 
-  m_addButtonOnClick(product: Product): void {
-    this.$appStore.cart.addProductToCart(product.id)
+  async m_addButtonOnClick(product: Product): Promise<void> {
+    await this.$appStore.cart.addProductToCart(product.id)
   }
 
   async m_checkoutButtonOnClick(): Promise<void> {
Index: package.json
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- package.json	(date 1548949595000)
+++ package.json	(date 1548951392000)
@@ -48,9 +48,9 @@
     "@vue/cli-plugin-pwa": "^3.2.0",
     "@vue/cli-plugin-typescript": "^3.2.0",
     "@vue/cli-service": "^3.2.0",
-    "browser-sync": "^2.26.3",
     "@vue/eslint-config-prettier": "^4.0.1",
     "@vue/eslint-config-typescript": "^3.2.1",
+    "browser-sync": "^2.26.3",
     "chai": "^4.2.0",
     "copy-webpack-plugin": "^4.6.0",
     "eslint": "^5.12.1",
Index: src/index.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/index.ts	(date 1548949595000)
+++ src/index.ts	(date 1548951392000)
@@ -9,13 +9,13 @@
 import { initAPI } from '@/apis'
 import { initConfig } from '@/base/config'
 import { initServiceWorker } from '@/base/service-worker'
-import { initStore } from '@/store'
+import { initStore, store } from '@/store'
 import { initUtils } from '@/base/utils'
 import { router } from '@/base/router'
 
 // TODO JetBrainsIDE使用時の開発補助コード
-// import '@/views/abc-view/index.vue'
-// import '@/views/shopping-view/index.vue'
+import '@/views/abc-view/index.vue'
+import '@/views/shopping-view/index.vue'
 
 (async () => {
   initUtils()
@@ -32,5 +32,6 @@
     router,
     render: (h) => h(AppView),
     i18n,
+    store,
   })
 })()
Index: src/store/index.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/store/index.ts	(date 1548949595000)
+++ src/store/index.ts	(date 1548951392000)
@@ -1,40 +1,31 @@
 import Vue from 'vue'
-import { newCartModule } from '@/store/cart-module'
-import { newProductModule } from '@/store/product-module'
-import { AppStore, CartModule, ProductModule } from '@/store/types'
-import { Component } from 'vue-property-decorator'
-
-const debug = process.env.NODE_ENV !== 'production'
+import Vuex, { Store, StoreOptions } from 'vuex'
 
-@Component
-class AppStoreImpl extends Vue implements AppStore {
-  constructor() {
-    super()
-    this.m_product = newProductModule()
-    this.m_cart = newCartModule({ product: this.product })
-  }
+import { CartTypes, CounterTypes, ProductTypes, RootState } from './types'
+import { newCartModule } from './cart-module'
+import { newCounterModule } from './counter-module'
+import { newProductModule } from './product-module'
+import { AppStore, newAppStore } from '@/store/app-store'
 
-  m_product: ProductModule
+Vue.use(Vuex)
 
-  get product(): ProductModule {
-    return this.m_product
-  }
-
-  m_cart: CartModule
-
-  get cart(): CartModule {
-    return this.m_cart
-  }
-}
+export const store = new Store<RootState>({
+  modules: {
+    [ProductTypes.PATH]: newProductModule(),
+    [CartTypes.PATH]: newCartModule(),
+    [CounterTypes.PATH]: newCounterModule(),
+  },
+} as StoreOptions<RootState>)
 
 export let appStore: AppStore
 
 export function initStore(): void {
-  appStore = new AppStoreImpl()
+  appStore = newAppStore(store)
   Object.defineProperty(Vue.prototype, '$appStore', {
     value: appStore,
     writable: false,
   })
 }
 
-export * from '@/store/types'
+export * from './types'
+export { AppStore }
Index: src/views/abc-view/index.vue
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/views/abc-view/index.vue	(date 1548949595000)
+++ src/views/abc-view/index.vue	(date 1548951392000)
@@ -58,7 +58,10 @@
           <span class="title">checked: </span><span class="value">{{ m_customChecked }}</span>
         </span>
       </div>
-      <div class="layout horizontal end-justified"><paper-button raised @click="m_sleepButtonOnClick">Sleep</paper-button></div>
+      <div class="layout horizontal end-justified">
+        <paper-button raised @click="m_incrementButtonOnClick">Increment</paper-button>
+        <paper-button raised @click="m_sleepButtonOnClick">Sleep</paper-button>
+      </div>
     </paper-card>
   </div>
 </template>
@@ -199,6 +202,11 @@
     alert(await this.m_sleep(2000))
   }
 
+  async m_incrementButtonOnClick() {
+    this.$appStore.counter.increment()
+    console.log(this.$appStore.counter.current)
+  }
+
   //--------------------------------------------------
   //  elements
   //--------------------------------------------------
Index: src/store/types.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/store/types.ts	(date 1548949595000)
+++ src/store/types.ts	(date 1548951392000)
@@ -1,4 +1,30 @@
-import { Product as APIProduct } from '@/apis'
+import { Module } from 'vuex'
+import { Product as APIProduct } from '../apis'
+
+//----------------------------------------------------------------------
+//
+//  States
+//
+//----------------------------------------------------------------------
+
+export interface RootState {
+  counter: CounterModule
+  product: ProductModule
+  cart: CartModule
+}
+
+export interface ProductState {
+  all: Product[]
+}
+
+export interface CartState {
+  items: Array<{ id: string, quantity: number }>
+  checkoutStatus: CheckoutStatus
+}
+
+export interface CounterState {
+  counter: number
+}
 
 //----------------------------------------------------------------------
 //
@@ -6,34 +32,62 @@
 //
 //----------------------------------------------------------------------
 
-export interface AppStore {
-  readonly product: ProductModule
+export interface ProductModule extends Module<ProductState, RootState> {}
+
+export interface CartModule extends Module<CartState, RootState> {}
+
+export interface CounterModule extends Module<CounterState, RootState> {}
+
+//----------------------------------------------------------------------
+//
+//  Types
+//
+//----------------------------------------------------------------------
+
+export namespace CounterTypes {
+  export const PATH = 'counter'
+
+  export const CURRENT = 'current'
 
-  readonly cart: CartModule
+  export const INCREMENT = 'increment'
 }
 
-export interface ProductModule {
-  readonly allProducts: Product[]
+export namespace ProductTypes {
+  export const PATH = 'product'
+
+  export const ALL_PRODUCTS = 'allProducts'
+
+  export const GET_PRODUCT_BY_ID = 'getProductById'
 
-  getProductById(productId: string): Product | undefined
+  export const SET_PRODUCTS = 'setProducts'
 
-  decrementProductInventory(productId: string): void
+  export const DECREMENT_PRODUCT_INVENTORY = 'decrementProductInventory'
 
-  pullAllProducts(): Promise<void>
+  export const PULL_ALL_PRODUCTS = 'pullAllProducts'
 }
 
-export interface CartModule {
-  readonly checkoutStatus: CheckoutStatus
+export namespace CartTypes {
+  export const PATH = 'cart'
+
+  export const CHECKOUT_STATUS = 'checkoutStatus'
+
+  export const CART_ITEMS = 'cartItems'
+
+  export const CART_TOTAL_PRICE = 'cartTotalPrice'
+
+  export const GET_CART_ITEM_BY_ID = 'getCartItemById'
 
-  readonly cartItems: CartItem[]
+  export const CHECKOUT = 'checkout'
 
-  readonly cartTotalPrice: number
+  export const ADD_PRODUCT_TO_CART = 'addProductToCart'
 
-  getCartItemById(productId: string): CartItem | undefined
+  export const m_SET_CART_ITEMS = 'm_setCartItems'
 
-  checkout(): Promise<void>
+  export const m_SET_CHECKOUT_STATUS = 'm_setCheckoutStatus'
 
-  addProductToCart(productId: string): void
+  export const m_PUSH_PRODUCT_TO_CART = 'm_pushProductToCart'
+
+  export const m_INCREMENT_ITEM_QUANTITY = 'm_incrementItemQuantity'
 }
 
 //----------------------------------------------------------------------
Index: vue.config.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- vue.config.js	(date 1548949595000)
+++ vue.config.js	(date 1548951392000)
@@ -19,12 +19,6 @@
 }
 if (process.env.VUE_APP_IS_DEVELOPMENT === 'true') {
   Object.assign(pages, {
-    test: {
-      entry: 'test/test.ts',
-      template: 'test/test.html',
-      filename: 'test.html',
-      title: 'Vue WWW Base Unit Test',
-    },
     playground: {
       entry: 'src/playground.ts',
       template: 'src/playground.html',
Index: src/store/cart-module/index.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/store/cart-module/index.ts	(date 1548949595000)
+++ src/store/cart-module/index.ts	(date 1548951392000)
@@ -1,155 +1,122 @@
-import { BaseModule } from '@/store/base'
-import { CartModule, CartItem, CheckoutStatus, Product, ProductModule } from '@/store/types'
-import { Component } from 'vue-property-decorator'
-import { NoCache } from '@/base/component'
-
-export interface CartState {
-  items: Array<{ id: string, quantity: number }>
-  checkoutStatus: CheckoutStatus
-}
+import { CartModule, CartState, RootState, CartItem, CartTypes, CheckoutStatus, Product, ProductState, ProductTypes } from '../types'
+import { GetterTree, MutationTree, ActionTree } from 'vuex'
+import { apis } from '@/apis'
 
-interface CartModuleDependencies {
-  product: ProductModule
-}
+class CartModuleImpl implements CartModule {
+  namespaced = true
 
-@Component
-export class CartModuleImpl extends BaseModule<CartState> implements CartModule {
-  //----------------------------------------------------------------------
-  //
-  //  Constructors
-  //
-  //----------------------------------------------------------------------
-
-  constructor() {
-    super()
-    this.f_initState({
-      items: [],
-      checkoutStatus: CheckoutStatus.None,
-    })
+  state: CartState = {
+    items: [],
+    checkoutStatus: CheckoutStatus.None,
   }
 
-  //----------------------------------------------------------------------
-  //
-  //  Variables
-  //
-  //----------------------------------------------------------------------
-
-  m_dependencies!: CartModuleDependencies;
-
-  //----------------------------------------------------------------------
-  //
-  //  Properties
-  //
-  //----------------------------------------------------------------------
-
-  get checkoutStatus(): CheckoutStatus {
-    return this.f_state.checkoutStatus
-  }
+  getters: GetterTree<CartState, RootState> = {
+    [CartTypes.CHECKOUT_STATUS](state): CheckoutStatus {
+      return state.checkoutStatus
+    },
 
-  @NoCache
-  get cartItems(): CartItem[] {
-    const allProducts = this.m_dependencies.product.allProducts
-    return this.f_state.items.map(({ id, quantity }) => {
-      const product = allProducts.find((item) => item.id === id)!
-      return {
-        id: product.id,
-        title: product.title,
-        price: product.price,
-        quantity,
-      }
-    })
-  }
+    [CartTypes.CART_ITEMS](state, getters, rootState): CartItem[] {
+      const allProducts = (rootState.product as ProductState).all
+      return state.items.map(({ id, quantity }) => {
+        const product = allProducts.find((item) => item.id === id)!
+        return {
+          id: product.id,
+          title: product.title,
+          price: product.price,
+          quantity,
+        }
+      })
+    },
 
-  get cartTotalPrice(): number {
-    return this.cartItems.reduce((total, product) => {
-      return total + product.price * product.quantity
-    }, 0)
-  }
-
-  //----------------------------------------------------------------------
-  //
-  //  Methods
-  //
-  //----------------------------------------------------------------------
+    [CartTypes.CART_TOTAL_PRICE](state, getters): number {
+      const cartItems = getters[CartTypes.CART_ITEMS] as CartItem[]
+      return cartItems.reduce((total, product) => {
+        return total + product.price * product.quantity
+      }, 0)
+    },
 
-  init(dependencies: CartModuleDependencies): void {
-    this.m_dependencies = dependencies
-  }
-
-  getCartItemById(productId: string): CartItem | undefined {
-    const product = this.m_getProductById(productId)
-    const cartItem = this.f_state.items.find((item) => {
-      return item.id === productId
-    })
-    if (!cartItem) return undefined
-    return {
-      id: cartItem.id,
-      title: product.title,
-      price: product.price,
-      quantity: cartItem.quantity,
-    }
+    [CartTypes.GET_CART_ITEM_BY_ID](state, getters, rootState, rootGetters): (productId: string) => CartItem | undefined {
+      return (productId: string) => {
+        const product = getProductById(rootGetters, productId)
+        const cartItem = state.items.find((item) => {
+          return item.id === productId
+        })
+        if (!cartItem) return undefined
+        return {
+          id: cartItem.id,
+          title: product.title,
+          price: product.price,
+          quantity: cartItem.quantity,
+        } as CartItem
+      }
+    },
   }
+
+  mutations: MutationTree<CartState> = {
+    [CartTypes.m_SET_CART_ITEMS](state, items: Array<{ id: string, quantity: number }>) {
+      state.items = items
+    },
+
+    [CartTypes.m_SET_CHECKOUT_STATUS](state, status: CheckoutStatus) {
+      state.checkoutStatus = status
+    },
+
+    [CartTypes.m_PUSH_PRODUCT_TO_CART](state, productId: string) {
+      state.items.push({
+        id: productId,
+        quantity: 1,
+      })
+    },
+
+    [CartTypes.m_INCREMENT_ITEM_QUANTITY](state, productId: string) {
+      const cartItem = state.items.find((item) => item.id === productId)
+      if (cartItem) {
+        cartItem.quantity++
+      }
+    },
+  }
 
-  addProductToCart(productId: string): void {
-    const product = this.m_getProductById(productId)
-    this.f_state.checkoutStatus = CheckoutStatus.None
-    if (product.inventory > 0) {
-      const cartItem = this.f_state.items.find((item) => item.id === product.id)
-      if (!cartItem) {
-        this.m_pushProductToCart(product.id)
-      } else {
-        this.m_incrementItemQuantity(cartItem.id)
-      }
-      // 在庫を1つ減らす
-      this.m_dependencies.product.decrementProductInventory(product.id)
-    }
-  }
+  actions: ActionTree<CartState, RootState> = {
+    async [CartTypes.ADD_PRODUCT_TO_CART](context, productId: string): Promise<void> {
+      context.commit(CartTypes.m_SET_CHECKOUT_STATUS, CheckoutStatus.None)
+      const product = getProductById(context.rootGetters, productId)
+      if (product.inventory > 0) {
+        const cartItem = context.state.items.find((item) => item.id === product.id)
+        if (!cartItem) {
+          context.commit(CartTypes.m_PUSH_PRODUCT_TO_CART, product.id)
+        } else {
+          context.commit(CartTypes.m_INCREMENT_ITEM_QUANTITY, product.id)
+        }
+        // 在庫を1つ減らす
+        context.commit(`${ProductTypes.PATH}/${ProductTypes.DECREMENT_PRODUCT_INVENTORY}`, productId, { root: true })
+      }
+    },
 
-  async checkout(): Promise<void> {
-    const savedCartItems = [ ...this.f_state.items ]
-    this.f_state.checkoutStatus = CheckoutStatus.None
-    try {
-      await this.$apis.shop.buyProducts(this.f_state.items)
-      this.f_state.items = [] // カートを空にする
-      this.f_state.checkoutStatus = CheckoutStatus.Successful
-    } catch (err) {
-      this.f_state.checkoutStatus = CheckoutStatus.Failed
-      // カートの内容をAPIリクエス前の状態にロールバックする
-      this.f_state.items = savedCartItems
-    }
+    async [CartTypes.CHECKOUT](context): Promise<void> {
+      const cartProducts = [ ...context.state.items ]
+      context.commit(CartTypes.m_SET_CHECKOUT_STATUS, CheckoutStatus.None)
+      try {
+        await apis.shop.buyProducts(cartProducts)
+        context.commit(CartTypes.m_SET_CART_ITEMS, []) // カートを空にする
+        context.commit(CartTypes.m_SET_CHECKOUT_STATUS, CheckoutStatus.Successful)
+      } catch (err) {
+        // カートの内容をAPIリクエス前の状態にロールバックする
+        context.commit(CartTypes.m_SET_CART_ITEMS, cartProducts)
+        context.commit(CartTypes.m_SET_CHECKOUT_STATUS, CheckoutStatus.Failed)
+      }
+    },
   }
-
-  //----------------------------------------------------------------------
-  //
-  //  Internal methods
-  //
-  //----------------------------------------------------------------------
-
-  m_pushProductToCart(productId: string): void {
-    this.f_state.items.push({
-      id: productId,
-      quantity: 1,
-    })
-  }
+}
 
-  m_incrementItemQuantity(productId: string): void {
-    const cartItem = this.f_state.items.find((item) => item.id === productId)
-    if (cartItem) {
-      cartItem.quantity++
-    }
-  }
-
-  m_getProductById(productId: string): Product {
-    const result = this.m_dependencies.product.getProductById(productId)
-    if (!result) {
-      throw new Error(`A product that matches the specified productId "${productId}" was not found.`)
-    }
-    return result
-  }
+function getProductById(rootGetters: any, productId: string): Product {
+  const path = `${ProductTypes.PATH}/${ProductTypes.GET_PRODUCT_BY_ID}`
+  const result = rootGetters[path](productId) as Product | undefined
+  if (!result) {
+    throw new Error(`A Product that matches the specified productId "${productId}" was not found.`)
+  }
+  return result
 }
 
-export function newCartModule(dependencies: CartModuleDependencies): CartModule {
-  const result = new CartModuleImpl()
-  result.init(dependencies)
-  return result
-}
+export function newCartModule() {
+  return new CartModuleImpl()
+}
\ No newline at end of file
