Index: src/views/product-admin-view/index.vue
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/views/product-admin-view/index.vue	(revision 03d309b7820d5160ac5983c326829a457770ac31)
+++ src/views/product-admin-view/index.vue	(revision 03d309b7820d5160ac5983c326829a457770ac31)
@@ -0,0 +1,14 @@
+<style lang="postcss" scoped></style>
+
+<template>
+  <div>product-admin-view</div>
+</template>
+
+<script lang="ts">
+import { Component } from 'vue-property-decorator';
+import { BaseComponent } from '../../base/component';
+import { mixins } from 'vue-class-component';
+
+@Component
+export default class ProductAdminView extends mixins(BaseComponent) {}
+</script>
Index: src/router/index.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/router/index.ts	(date 1544028305000)
+++ src/router/index.ts	(date 1544028366000)
@@ -14,5 +14,9 @@
       path: '/shopping',
       component: () => import(/* webpackChunkName: "shopping-view" */ '../views/shopping-view/index.vue'),
     },
+    {
+      path: '/product-admin',
+      component: () => import(/* webpackChunkName: "product-admin-view" */ '../views/product-admin-view/index.vue'),
+    },
   ],
 });
Index: src/views/index.vue
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/views/index.vue	(date 1544028305000)
+++ src/views/index.vue	(date 1544028366000)
@@ -104,6 +104,10 @@
       title: 'Shopping',
       path: '/shopping',
     },
+    {
+      title: 'Product Admin',
+      path: '/product-admin',
+    },
   ];
 
   m_swMessage: string = '';
