Index: src/views/product-admin-view/index.vue
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/views/product-admin-view/index.vue	(date 1544030832000)
+++ src/views/product-admin-view/index.vue	(date 1544030832000)
@@ -0,0 +1,225 @@
+<style scoped>
+@import '../../styles/flex-layout.css';
+@import '../../styles/typography.css';
+
+.title-text {
+  @extend %comm-font-title;
+}
+
+.product-item {
+  padding: 12px;
+  cursor: pointer;
+
+  &:hover {
+    background-color: var(--app-grid-selected-hover-item);
+  }
+
+  &.iron-selected {
+    background-color: var(--app-grid-selected-activ-item);
+  }
+
+  & .title {
+    @extend %comm-font-subhead;
+  }
+
+  & .detail {
+    @extend %comm-font-body1;
+    color: var(--app-secondary-text-color);
+  }
+}
+
+.product-detail {
+  & .row {
+    width: 500px;
+    &.sp {
+      width: 100%;
+    }
+  }
+
+  & .title-area {
+    @extend %comm-font-subhead;
+    @extend %layout-flex-2;
+    margin-top: 16px;
+  }
+
+  & .input-area {
+    @extend %layout-flex-10;
+
+    & .title-input {
+      width: 400px;
+      &.sp {
+        width: 100%;
+      }
+    }
+
+    & .price-input,
+    & .inventory-input {
+      width: 100px;
+      text-align: right;
+    }
+  }
+}
+
+.update-button {
+  color: var(--app-link-color);
+}
+</style>
+
+<template>
+  <div class="layout vertical" :class="{ 'comm-ma-48': f_pc, 'comm-ma-24': f_tab, 'comm-ma-12': f_sp }">
+    <div>
+      <div class="layout horizontal center">
+        <div class="title-text">{{ $t('products') }}</div>
+      </div>
+      <hr style="width: 100%;" />
+      <iron-selector ref="productSelector" attr-for-selected="product-id" @selected-changed="m_productSelectorOnSelectedItemChanged">
+        <div v-for="(product, index) in m_products" class="layout horizontal center product-item" :product-id="product.id">
+          <div class="layout vertical center-justified">
+            <div class="title">{{ product.title }}</div>
+            <div class="detail">
+              <span>Price</span> &mdash; {{ product.price | currency }},&nbsp; <span>Stock</span> &mdash; {{ product.inventory }}
+            </div>
+          </div>
+          <div class="flex"></div>
+        </div>
+      </iron-selector>
+    </div>
+
+    <div class="comm-mt-20">
+      <div class="layout horizontal center">
+        <div class="title-text">{{ $t('detail') }}</div>
+      </div>
+      <hr style="width: 100%;" />
+      <div class="layout vertical product-detail">
+        <div class="layout horizontal center row" :class="{ sp: f_sp }">
+          <div class="title-area">{{ $t('title') }}</div>
+          <div class="input-area">
+            <paper-input
+              class="title-input"
+              :class="{ sp: f_sp }"
+              :value="m_selectedProduct.title"
+              @input="m_selectedProduct.title = $event.target.value;"
+            ></paper-input>
+          </div>
+        </div>
+        <div class="layout horizontal center row" :class="{ sp: f_sp }">
+          <div class="title-area">{{ $t('price') }}</div>
+          <div class="input-area">
+            <paper-input
+              class="price-input"
+              type="number"
+              :value="m_selectedProduct.price"
+              @input="m_selectedProduct.price = $event.target.value;"
+              @change="m_selectedProduct.price = $event.target.value;"
+            >
+              <div slot="prefix">$</div>
+            </paper-input>
+          </div>
+        </div>
+        <div class="layout horizontal center row" :class="{ sp: f_sp }">
+          <div class="title-area">{{ $t('stock') }}</div>
+          <div class="input-area">
+            <paper-input
+              class="inventory-input"
+              type="number"
+              :value="m_selectedProduct.inventory"
+              @input="m_selectedProduct.inventory = $event.target.value;"
+              @change="m_selectedProduct.inventory = $event.target.value;"
+            ></paper-input>
+          </div>
+        </div>
+      </div>
+      <div class="layout horizontal center end-justified">
+        <paper-button v-show="!!m_selectedProduct.id" class="update-button" @click="m_updateProduct"> {{ $t('update') }} </paper-button>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import '@polymer/iron-selector/iron-selector';
+import '@polymer/paper-input/paper-input';
+
+import { Component } from 'vue-property-decorator';
+import { BaseComponent } from '../../base/component';
+import { Product } from '../../stores';
+import { mixins } from 'vue-class-component';
+
+@Component
+export default class ProductAdminView extends mixins(BaseComponent) {
+  //----------------------------------------------------------------------
+  //
+  //  Variables
+  //
+  //----------------------------------------------------------------------
+
+  m_selectedProduct: Product | object = {};
+
+  get m_products(): Product[] {
+    return this.$stores.product.allProducts;
+  }
+
+  //--------------------------------------------------
+  //  Elements
+  //--------------------------------------------------
+
+  get m_productSelector(): {
+    selected: string;
+  } {
+    return this.$refs.productSelector as any;
+  }
+
+  //----------------------------------------------------------------------
+  //
+  //  Lifecycle hooks
+  //
+  //----------------------------------------------------------------------
+
+  created() {}
+
+  //----------------------------------------------------------------------
+  //
+  //  Internal methods
+  //
+  //----------------------------------------------------------------------
+
+  async m_updateProduct(): Promise<void> {
+    await this.$stores.product.updateProduct(this.m_selectedProduct as Product);
+  }
+
+  //----------------------------------------------------------------------
+  //
+  //  Event handlers
+  //
+  //----------------------------------------------------------------------
+
+  m_productSelectorOnSelectedItemChanged(event) {
+    let selectedProduct: Product | undefined;
+    if (this.m_productSelector.selected) {
+      const productId = this.m_productSelector.selected;
+      const foundProduct = this.m_products.find((item) => item.id === productId);
+      selectedProduct = this.$utils.cloneDeep(foundProduct);
+    }
+    // tslint:disable-next-line
+    console.log('selectedProduct:', selectedProduct);
+    this.m_selectedProduct = selectedProduct || {};
+  }
+}
+</script>
+
+<i18n>
+en:
+  products: "Products"
+  detail: "Detail"
+  title: "Title"
+  price: "Price"
+  stock: "Stock"
+  update: "Update"
+ja:
+  products: "商品一覧"
+  detail: "詳細"
+  title: "タイトル"
+  price: "価格"
+  stock: "在庫"
+  update: "更新"
+</i18n>
Index: src/stores/types.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/stores/types.ts	(date 1544030764000)
+++ src/stores/types.ts	(date 1544030832000)
@@ -20,6 +20,8 @@
   decrementProductInventory(productId: string): void;
 
   getAllProducts(): Promise<void>;
+
+  updateProduct(product: Product): Promise<void>;
 }
 
 export interface CartStore {
Index: src/stores/product-store/index.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/stores/product-store/index.ts	(date 1544030764000)
+++ src/stores/product-store/index.ts	(date 1544030832000)
@@ -66,6 +66,14 @@
     this.f_state.all = products;
   }
 
+  async updateProduct(product: Product): Promise<void> {
+    const stateProduct = this.m_getStateProductById(product.id);
+    if (stateProduct) {
+      await this.$apis.shop.updateProduct(product);
+      this.$utils.assignIn(stateProduct, product);
+    }
+  }
+
   //----------------------------------------------------------------------
   //
   //  Internal methods
Index: src/apis/types.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/apis/types.ts	(date 1544030764000)
+++ src/apis/types.ts	(date 1544030832000)
@@ -12,6 +12,8 @@
   getProducts(): Promise<Product[]>;
 
   buyProducts(products: Array<{ id: string; quantity: number }>): Promise<void>;
+
+  updateProduct(product: Product): Promise<void>;
 }
 
 //----------------------------------------------------------------------
Index: src/router/index.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/router/index.ts	(date 1544030764000)
+++ src/router/index.ts	(date 1544030801000)
@@ -14,5 +14,9 @@
       path: '/shopping',
       component: () => import(/* webpackChunkName: "shopping-view" */ '../views/shopping-view/index.vue'),
     },
+    {
+      path: '/product-admin',
+      component: () => import(/* webpackChunkName: "product-admin-view" */ '../views/product-admin-view/index.vue'),
+    },
   ],
 });
Index: src/apis/shop-api/index.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/apis/shop-api/index.ts	(date 1544030764000)
+++ src/apis/shop-api/index.ts	(date 1544030832000)
@@ -17,6 +17,10 @@
       }, 100);
     });
   }
+
+  async updateProduct(product: Product): Promise<void> {
+    await this.put(`products/${product.id}`, product);
+  }
 }
 
 const shopAPI: ShopAPI = new ShopAPIImpl();
Index: test/stores/product-store/index.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- test/stores/product-store/index.ts	(date 1544030764000)
+++ test/stores/product-store/index.ts	(date 1544030844000)
@@ -61,4 +61,39 @@
     await productStore.getAllProducts();
     assert.deepEqual(productStore.allProducts, NEW_API_PRODUCTS);
   });
+
+  test('updateProduct() - 一般ケース', async () => {
+    // 対象の商品に更新データを設定
+    const PRODUCT_ID = '1';
+    const product = productStore.getProductById(PRODUCT_ID);
+    if (!product) throw new Error(`Product \`${PRODUCT_ID}\` was not found.`);
+    product.title = 'aaa';
+    product.price = 111;
+    product.inventory = 222;
+
+    // モック処理の設定
+    const stateProduct = {} as Product;
+    td.replace(productStore, 'm_getStateProductById');
+    td.when(productStore.m_getStateProductById(PRODUCT_ID)).thenReturn(stateProduct);
+    td.replace(shopAPI, 'updateProduct');
+
+    // 商品の更新を実行
+    await productStore.updateProduct(product);
+
+    // Stateの商品と引数の商品の内容が一致することを検証
+    // (Stateの商品に引数の商品の内容が反映されるため)
+    assert.deepEqual(stateProduct, product);
+    // `ShopAPI#updateProduct()`が呼び出されたことを検証
+    td.verify(shopAPI.updateProduct(product));
+  });
+
+  test('updateProduct() - 存在しない商品を指定した場合', async () => {
+    // 対象の商品に更新データを設定
+    const product = productStore.getProductById('1') as Product;
+    product.id = '9876';
+    // 商品の更新を実行
+    await productStore.updateProduct(product);
+    // 何も問題は起きない
+    assert(true);
+  });
 });
Index: src/index.css
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/index.css	(date 1544030764000)
+++ src/index.css	(date 1544030816000)
@@ -18,6 +18,8 @@
   --app-link-color: var(--comm-indigo-500);
   --app-link-visited-color: var(--comm-indigo-300);
   --app-default-border-color: var(--comm-grey-300);
+  --app-grid-selected-hover-item: var(--comm-grey-100);
+  --app-grid-selected-activ-item: var(--comm-grey-300);
 
   /*------------------------------
     Grid
Index: src/views/index.vue
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/views/index.vue	(date 1544030764000)
+++ src/views/index.vue	(date 1544030801000)
@@ -104,6 +104,10 @@
       title: 'Shopping',
       path: '/shopping',
     },
+    {
+      title: 'Product Admin',
+      path: '/product-admin',
+    },
   ];
 
   m_swMessage: string = '';
