Index: src/index.css
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/index.css	(date 1544028509000)
+++ src/index.css	(date 1544028866000)
@@ -18,6 +18,8 @@
   --app-link-color: var(--comm-indigo-500);
   --app-link-visited-color: var(--comm-indigo-300);
   --app-default-border-color: var(--comm-grey-300);
+  --app-grid-selected-hover-item: var(--comm-grey-100);
+  --app-grid-selected-activ-item: var(--comm-grey-300);
 
   /*------------------------------
     Grid
Index: src/views/product-admin-view/index.vue
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/views/product-admin-view/index.vue	(date 1544028509000)
+++ src/views/product-admin-view/index.vue	(date 1544028866000)
@@ -7,6 +7,15 @@
 
 .product-item {
   padding: 12px;
+  cursor: pointer;
+
+  &:hover {
+    background-color: var(--app-grid-selected-hover-item);
+  }
+
+  &.iron-selected {
+    background-color: var(--app-grid-selected-activ-item);
+  }
 
   & .title {
     @extend %comm-font-subhead;
@@ -26,21 +35,24 @@
         <div class="title-text">{{ $t('products') }}</div>
       </div>
       <hr style="width: 100%;" />
-      <div v-for="(product, index) in m_products" class="layout horizontal center product-item">
-        <div class="layout vertical center-justified">
-          <div class="title">{{ product.title }}</div>
-          <div class="detail">
-            <span>{{ $t('price') }}</span> &mdash; {{ product.price | currency }},&nbsp; <span>{{ $t('stock') }}</span> &mdash;
-            {{ product.inventory }}
-          </div>
-        </div>
-        <div class="flex"></div>
-      </div>
+      <iron-selector ref="productSelector" attr-for-selected="product-id" @selected-changed="m_productSelectorOnSelectedItemChanged">
+        <div v-for="(product, index) in m_products" class="layout horizontal center product-item" :product-id="product.id">
+          <div class="layout vertical center-justified">
+            <div class="title">{{ product.title }}</div>
+            <div class="detail">
+              <span>Price</span> &mdash; {{ product.price | currency }},&nbsp; <span>Stock</span> &mdash; {{ product.inventory }}
+            </div>
+          </div>
+          <div class="flex"></div>
+        </div>
+      </iron-selector>
     </div>
   </div>
 </template>
 
 <script lang="ts">
+import '@polymer/iron-selector/iron-selector';
+
 import { Component } from 'vue-property-decorator';
 import { BaseComponent } from '../../base/component';
 import { Product } from '../../stores';
@@ -58,6 +70,16 @@
     return this.$stores.product.allProducts;
   }
 
+  //--------------------------------------------------
+  //  Elements
+  //--------------------------------------------------
+
+  get m_productSelector(): {
+    selected: string;
+  } {
+    return this.$refs.productSelector as any;
+  }
+
   //----------------------------------------------------------------------
   //
   //  Lifecycle hooks
@@ -65,6 +87,22 @@
   //----------------------------------------------------------------------
 
   created() {}
+
+  //----------------------------------------------------------------------
+  //
+  //  Event handlers
+  //
+  //----------------------------------------------------------------------
+
+  m_productSelectorOnSelectedItemChanged(event) {
+    let selectedProduct: Product | undefined;
+    if (this.m_productSelector.selected) {
+      const productId = this.m_productSelector.selected;
+      selectedProduct = this.m_products.find((item) => item.id === productId);
+    }
+    // tslint:disable-next-line
+    console.log('selectedProduct:', selectedProduct);
+  }
 }
 </script>
 
