Index: src/views/product-admin-view/index.vue
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/views/product-admin-view/index.vue	(revision 63b6e3dbea2b7b823161318b4544ee3c0ac57016)
+++ src/views/product-admin-view/index.vue	(revision 63b6e3dbea2b7b823161318b4544ee3c0ac57016)
@@ -0,0 +1,14 @@
+<style lang="postcss" scoped></style>
+
+<template>
+  <div>product-admin-view</div>
+</template>
+
+<script lang="ts">
+import { Component } from 'vue-property-decorator';
+import { BaseComponent } from '../../base/component';
+import { mixins } from 'vue-class-component';
+
+@Component
+export default class ProductAdminView extends mixins(BaseComponent) {}
+</script>
Index: src/router/index.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/router/index.ts	(date 1541858947000)
+++ src/router/index.ts	(date 1541860055000)
@@ -15,5 +15,10 @@
       component: () =>
         import(/* webpackChunkName: "shopping-view" */ '../views/shopping-view/index.vue'),
     },
+    {
+      path: '/product-admin',
+      component: () =>
+        import(/* webpackChunkName: "product-admin-view" */ '../views/product-admin-view/index.vue'),
+    },
   ],
 });
Index: src/views/index.vue
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/views/index.vue	(date 1541858947000)
+++ src/views/index.vue	(date 1541860055000)
@@ -106,6 +106,10 @@
       title: 'Shopping',
       path: '/shopping',
     },
+    {
+      title: 'Product Admin',
+      path: '/product-admin',
+    },
   ];
 
   m_swMessage: string = '';
